core/shared/src/main/scala/spire/algebra/DivisionRing.scala

```diff
@@ -0,0 +1,50 @@
+package spire
+package algebra
+
+import java.lang.Double.{ isInfinite, isNaN, doubleToLongBits }
+import java.lang.Long.{ numberOfTrailingZeros }
+
+trait DivisionRing[@sp(Byte, Short, Int, Long, Float, Double) A] extends Any with Ring[A] with MultiplicativeGroup[A] { self =>
+
+  def fromDouble(a: Double): A = DivisionRing.fromDouble[A](a)(self, self)
+}
+
+
+object DivisionRing {
+
+  @inline final def apply[A](implicit f: DivisionRing[A]): DivisionRing[A] = f
+
+  // TODO: use instead algebra Ring.fromDouble, if PR algebra/#190 is accepted
```

Comment:
typelevel/algebra#190 was merged but not published. Can Spire depend on snapshot versions of algebra?